server/circleci.go

```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/metanerd/go-circleci"
 )
 
-func (s *Server) triggerCircleCiIfNeeded(pr *model.PullRequest) {
+func (s *Server) triggerCircleCiIfNeeded(ctx context.Context, pr *model.PullRequest) {
 	client := &circleci.Client{Token: s.Config.CircleCIToken}
```

Comment:
We also need to pass timeouts to the circleCI calls. The metanerd/go-circleci package does not expose context in the API calls. I am 0/5 on whether we want to bring the package inside this repo or modify it upstream. But let's file a ticket to pass the context.

Reply:
That project is also a fork from Carlos's fork :laughing: . I'll ask here to open issues and fill one asking for exposing the context in the calls

Reply:
Thanks :)

Reply:
Create https://mattermost.atlassian.net/browse/MM-26589